### PR TITLE
Update time dependent bcs

### PR DESCRIPTION
**Pull request by jhdark**
## Proposed changes

Currently, all boundary conditions are updated at each time step to ensure the value is updated in case it's dependent on time. With this change, only boundary conditions with values dependent on time will be updated at each time step. This should also improve performance.

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [x] Black formatted
- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by RemDelaporteMathurin** [comment]
Let's wait until #620 is merged before looking at this

**Review by RemDelaporteMathurin — APPROVED** [review]
This looks good to me! I know it's hard to test that the bc is updated only once.
Although we could potentially do it by timing the operation, but maybe a manual test is enough for now.

Have you tested it manually by having print statements?

**Comment by jhdark** [comment]
> This looks good to me! I know it's hard to test that the bc is updated only once. Although we could potentially do it by timing the operation, but maybe a manual test is enough for now.
> 
> Have you tested it manually by having print statements?

Yep, I'll merge it then

**Pull request merged**
